File: algorithms/searchs/binary_search.py
def binary_search(list: list, target: int) -> int | None:
    """Returns the index position of the target if found, else returns None"""
    first = 0
    last = len(list) - 1

    while first <= last:
        mid = (first + last) // 2 # Floor division
        if list[mid] == target:
            return mid
        elif list[mid] < target:
            first = mid + 1
        else:
            last = mid - 1
    return None

def verify(index):
    if index is not None:
        print("Target found at index: ", index)
    else:
        print("Target not found in list")

numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
result = binary_search(numbers, 7)
verify(result)

File: algorithms/searchs/binarySearch.js

function binarySearch(nums, search) {
  if (nums.length === 0) {
    console.log('There is no ' + search + ' in the list of numbers');
    return false;
  }
  const mid = Math.floor(nums.length / 2);
  console.log(`Is ${search} at the mid(${mid}) of ${nums}?`);
  if (nums[mid] > search) {
    console.log('No');
    return binarySearch(nums.slice(0, mid), search);
  } else if (nums[mid] < search) {
    console.log('No');
    return binarySearch(nums.slice(mid + 1), search);
  } else {
    console.log('Yes');
    return true;
  }
};
const nums = [1, 2, 3, 4, 5, 6, 8, 9, 10, 11, 12, 13, 14, 15, 16];

binarySearch(nums, 3);
binarySearch(nums, 8);
binarySearch(nums, 14);
binarySearch(nums, 7);
